lib/phoenix_live_view/async.ex

```diff
@@ -3,11 +3,71 @@ defmodule Phoenix.LiveView.Async do
 
   alias Phoenix.LiveView.{AsyncResult, Socket, Channel}
 
-  def start_async(%Socket{} = socket, key, func, opts \\ []) when is_function(func, 0) do
-    run_async_task(socket, key, func, :start, opts)
+  defp validate_function_env(func, op, env) do
+    # TODO: is there a better way to check if socket is used in func?
```

Comment:
That's the way to go do it. But it can yield false positives. For example:

    start_async(function_that_returns_the_anonymous_function(socket))

So it may be better to check if `func` matches either `{:fn, _, _}` or `{:&, _, _}`.